pkg/controller/instanceset/reconciler_update.go

```diff
@@ -142,6 +142,14 @@ func (r *updateReconciler) Reconcile(tree *kubebuilderx.ObjectTree) (*kubebuilde
 		if err != nil {
 			return nil, err
 		}
+		if its.Spec.PodUpdatePolicy == workloads.StrictInPlacePodUpdatePolicyType && updatePolicy == RecreatePolicy {
+			if tree != nil && tree.EventRecorder != nil {
+				tree.EventRecorder.Eventf(its, corev1.EventTypeWarning, EventReasonStrictInPlace,
+					fmt.Sprintf("InstanceSet %s/%s blocks on scale-in as the PodUpdatePolicy is %s and the pod %s can not inplace update",
+						its.Namespace, its.Name, workloads.StrictInPlacePodUpdatePolicyType, pod.Name))
+			}
+			break
```

Comment:
Should we leave some message (label, annotation or condition, to be discussed) on its ?

Reply:
> Should we leave some message (label, annotation or condition, to be discussed) on its ?

Yes, it seems that having a condition on the instanceSet would be better.